docs/_docs/liquid/tags.md

```diff
@@ -57,15 +60,61 @@ numbers. For instance, the following code block would include line numbers next
 to each line:
 
 {% raw %}
+
 ```liquid
 {% highlight ruby linenos %}
 def foo
   puts 'foo'
 end
 {% endhighlight %}
 ```
+
+{% endraw %}
+
+### Variables for language and line numbers
+
+{: .note}
+To pass a language variable to the `highlight` tag, the variable **must** be named `lang`, as in {% raw %}{{ page.lang }}{% endraw %}. The `linenos` variable can be called by any other name.
```

Comment:
I don't think this statement is true about _must_ use "lang".

Variable names don't matter. the value matters


Have you tried using another name and see what happens?

```
---
language: ruby
---
{% highlight {{ page.language }} %}
...
```

Reply:
Thanks for your comments.

In order to differentiate between the `lang` and `linenos` variables I used a regexp that catches only variable names that end in 'lang'.

Testing the Regexp [here](https://rubular.com/r/fCGFAF7dCFA90E) with `{{ page.language }}` catches the variable in the parameters (linenos) var. But I think it would be useful to catch all similar variables (lang / language / lang_a / lang1 etc.) so that several top-level language variables could be specified for the page, like so:
```text
---
lang1: cpp
lang2: html
---
{% highlight {{ page.lang1 }} %}
/* code */
{% endhighlight %}
...
{% highlight {{ page.lang2 }} %}
<!-- markup -->
{% endhighlight %}
```
I tried to accommodate that by expanding the regexp a bit, and would appreciate testing. [Here](https://rubular.com/r/riHdOEUdsaaFAT) is the revised regexp.

Reply:
Oh okay that makes sensitive but sounds like an anti-pattern in programming. I didn't know the tag could even know the name of the field passed to it, I thought it would use the evaluated value only.

I would have expected something like you pass one parameter and that is only lang, or pass two and that is lang and then linenos, exactly in that order only (no  regex needed then as you have positional parameters).

If there was a concern about the need to use no land but still set linenos, I would have expected  calling as `{% highlight nil true %}`. Or using named parameters - if tags support this like filters do? `{% highlight linenos=true %}` and  `{% highlight {{ page.lang }} linenos=true %}`